src/routing_membrane.rs

```diff
@@ -514,6 +519,26 @@ impl<F> RoutingMembrane<F> where F: Interface {
         }
     }
 
+    fn send_by_name<'a, I>(&self, peers: I, serialised_msg: Bytes) where I: Iterator<Item=&'a NameType> {
+        for peer in peers {
+            self.send(self.name_to_endpoint(peer).into_iter(), &serialised_msg);
+        }
+    }
+
+    fn name_to_endpoint(&self, name: &NameType) -> Vec<&Endpoint> {
```

Comment:
relay_map already has a lookup function; so has routing_table.  So depending on where you call it; you should know from the context that you only want to search through relay map exclusive or routing_table.  From the message handlers you should only ever search the routing table and forward inside the SAFE network.  No handler should send to a relay endpoint - I wrote all those